### PR TITLE
F #sp 476 vs code setup lint

### PR DESCRIPTION
**Pull request by ethanfowler**
## Proposed changes

Adding VS code config/common developer workflow. Branching from `lint` branch as we seem to be using that as CI trunk.
## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [x] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Comment by toliver** [comment]
Pretty weird that we have `master` and `lint` branches. Both having exclusive commits.

**Comment by ethanfowler** [comment]
> Pretty weird that we have `master` and `lint` branches. Both having exclusive commits.

I agree. Not sure why they diverged historically. It's going to take some effort to unpick.

File: ansible/roles/ci/code_style_check/license/files/copyright_checks.py [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+import os
+import re
+import subprocess
+import sys
+
+# We want to use f-strings for logging. Very minor performance hit, better consistency.
+# pylint: disable=logging-fstring-interpolation
+
+
+class CopyrightChecker:
```
Comment (toliver) [inline ansible/roles/ci/code_style_check/license/files/copyright_checks.py:29]: I'm a bit confused. I thought this functionality was already part of our CI infrastructure.
Is this a duplication or a copy of scripts that we have elsewhere?

File: ansible/roles/ci/code_style_check/python/files/pylintrc [hunk 2]
```diff
@@ -114,7 +114,7 @@ score=no
 [REFACTORING]
 
 # Maximum number of nested blocks for function / method body
-max-nested-blocks=5
+max-nested-blocks=7
```
[the remark this answers is not in the document]
Comment (toliver) [reply ansible/roles/ci/code_style_check/python/files/pylintrc:117]: Oh I see it was defined as 7 further down

File: ansible/roles/dev_machine/files/vs_code_setup.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+sr_build_tools_branch="F_%23SP-476_vs_code_setup_lint"
```
Comment (toliver) [inline ansible/roles/dev_machine/files/vs_code_setup.sh:17]: I guess we would need to change this before we can merge.

File: ansible/roles/ros_workspace/files/vs_code/one_line_install.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+build_tools_dir=~/sr-build-tools
+build_tools_branch=F_#SP-476_vs_code_setup_lint
```
Comment (toliver) [inline ansible/roles/ros_workspace/files/vs_code/one_line_install.sh:18]: Likewise. We should probably point to master before we merge.

Comment (ethanfowler) [reply ansible/roles/ros_workspace/files/vs_code/one_line_install.sh]: Addressed in https://github.com/shadow-robot/sr-build-tools/pull/697/commits/89c8670d4f1388b9f4219295f7f553aa6b8b76ed

File: ansible/roles/dev_machine/files/vs_code_setup.sh (hunk 3 above)
Comment (ethanfowler) [reply ansible/roles/dev_machine/files/vs_code_setup.sh]: Addressed in https://github.com/shadow-robot/sr-build-tools/pull/697/commits/89c8670d4f1388b9f4219295f7f553aa6b8b76ed

File: ansible/roles/ci/code_style_check/catkin/files/catkin_lint_local.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+for match in re.findall(regex, remaining_output, re.MULTILINE):
+    package_path = get_package_path(match[1])
+    if match[3] == '':
+        print(f'{package_path}/{match[2]}:0:{match[4]}:{match[5]}')
+    else:
+        print(f'{package_path}/{match[2]}:{match[3]}:{match[4]}:{match[5]}')
+    remaining_output = remaining_output.replace(f'\n{match[0]}', '')
+
+regex = r"(^(.*?):\s(.*?): (file '(.*?)'.*)$)"
```
[the remark this answers is not in the document]
Comment (ethanfowler) [reply ansible/roles/ci/code_style_check/catkin/files/catkin_lint_local.py:63]: Addressed in https://github.com/shadow-robot/sr-build-tools/pull/697/commits/89c8670d4f1388b9f4219295f7f553aa6b8b76ed

**Review by toliver — APPROVED** [review]

**Pull request merged**